pyradiosky/skymodel.py

```diff
@@ -108,6 +105,177 @@ def _get_matching_fields(
+        frame_desc_str = frame_obj.name
+    elif frame_obj.name == "fk5" and frame_obj.equinox == Time("j2000"):
+        # this is for backwards compatibility and standards in the literature
+        # coordinates reported in J2000 implicitly means FK5 unless otherwise stated
+        frame_desc_str = "j2000"
+    elif frame_obj.name == "fk4" and frame_obj.equinox == Time("b1950"):
+        # this is for backwards compatibility and standards in the literature
+        # coordinates reported in B1950 implicitly means FK4 unless otherwise stated
+        frame_desc_str = "b1950"
+    elif frame_obj.name == "fk5":
+        frame_desc_str = frame_obj.name + "_" + frame_obj.equinox.jyear_str
+    elif frame_obj.name == "fk4":
+        frame_desc_str = frame_obj.name + "_" + frame_obj.equinox.byear_str
+    else:
+        raise ValueError(f"{frame_obj.name} is not supported for writing text files.")
+    return frame_desc_str
+
+
+def _get_frame_comp_cols(colnames):
+    frame_use = None
+    frame_names = frame_transform_graph.get_names()
+    lon_col = None
+    lat_col = None
+    ra_fk5_pattern = re.compile("ra[_]?j2000")
+    dec_fk5_pattern = re.compile("de[c]?[_]?j2000")
+    ra_fk4_pattern = re.compile("ra[_]?b1950")
+    dec_fk4_pattern = re.compile("de[c]?[_]?b1950")
+    for name in colnames:
+        if (
+            ra_fk5_pattern.match(name.casefold()) is not None
+            or dec_fk5_pattern.match(name.casefold()) is not None
+        ):
+            frame_use = SkyCoord(0, 0, unit="deg", frame="fk5").frame
+            break
+        elif (
+            ra_fk4_pattern.match(name.casefold()) is not None
+            or dec_fk4_pattern.match(name.casefold()) is not None
+        ):
+            frame_use = SkyCoord(0, 0, unit="deg", frame="fk4").frame
+            break
+
+        for frame in frame_names:
+            if frame in name.casefold():
+                frame_use = frame
+                break
+        if frame_use is not None:
+            break
+
+    if isinstance(frame_use, str):
+        default_skycoord = SkyCoord(0, 0, unit="deg", frame=frame_use)
+        lon_name, lat_name = _get_lon_lat_component_names(default_skycoord)
+        for name in colnames:
+            if lon_name in name.casefold():
+                lon_col = name
+            if lat_name in name.casefold():
+                lat_col = name
+            if lon_col is not None and lat_col is not None:
+                break
+        if lon_col is not None:
+            frame_str = lon_col.casefold().split(lon_name + "_", 1)[1]
+        elif lat_col is not None:
+            frame_str = lat_col.casefold().split(lat_name + "_", 1)[1]
+
+        if frame_str == frame_use:
+            frame_use = default_skycoord.frame
+        elif frame_use in ["fk4", "fk5"]:
+            if frame_use == "fk4":
+                equinox = Time("b" + frame_str.split("_b", 1)[1])
+            else:
+                equinox = Time("j" + frame_str.split("_j", 1)[1])
```

Comment:
earlier we changed to a `[_]?` regex to allow 0 or 1 underscore characters, but these lines trying to find the equinox assume a `_[bj]` pattern. Just wondering if we're going to code ourselves into a corner here.

Reply:
Yep, good point. I'm going to require exactly 1 underscore.